File: app/src/main/java/com/aganticsoft/phyxhidephotosandvideos/di/module/ActivityBindingModule.java
package com.aganticsoft.phyxhidephotosandvideos.di.module;

import com.aganticsoft.phyxhidephotosandvideos.di.scope.ActivityScope;
import com.aganticsoft.phyxhidephotosandvideos.view.activity.MainActivity;
import com.aganticsoft.phyxhidephotosandvideos.view.activity.MediaChooseActivity;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

/**
 * Created by ttson
 * Date: 6/18/2017.
 */
@Module
public abstract class ActivityBindingModule {

    @ActivityScope
    @ContributesAndroidInjector(modules = {MainFragmentModule.class})
    abstract MainActivity contributeMainActivty();

    @ActivityScope
    @ContributesAndroidInjector(modules = {MediaChooseFragmentModule.class})
    abstract MediaChooseActivity contributeMediaChooseActivity();
}

File: app/src/main/java/com/aganticsoft/phyxhidephotosandvideos/di/module/ActivityBindingModule.kt
package com.aganticsoft.phyxhidephotosandvideos.di.module;

import com.aganticsoft.phyxhidephotosandvideos.MainActivity
import com.aganticsoft.phyxhidephotosandvideos.di.scope.ActivityScope;

import dagger.Module
import dagger.android.ContributesAndroidInjector

/**
 * Created by ttson
 * Date: 6/18/2017.
 */
@Module
abstract class ActivityBindingModule {

    @ActivityScope
    @ContributesAndroidInjector(modules = arrayOf())
    abstract fun contributeMainActivty() : MainActivity
}